lib.php

```diff
@@ -663,11 +663,11 @@ function diary_print_overview($courses, &$htmlarray) {
         }
 
         if ($diaryopen) {
+            $url = new moodle_url($CFG->wwwroot . '/mod/diary/view.php', array('id' => $diary->coursemodule));
```

Comment:
You generally dont need to declare the wwwroot in a moodle_url constructure iirc, it defaults to that, so you can just create the URL with the application path